src/github.com/getlantern/yamlconf/yamlconf_disk.go

```diff
@@ -55,6 +78,37 @@ func (m *Manager) reloadFromDisk() (bool, error) {
 	return true, nil
 }
 
+func (m *Manager) doReadFromDisk(allowObfuscation bool) (Config, error) {
+	infile, err := os.Open(m.FilePath)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to open config file %v for reading: %v", m.FilePath, err)
+	}
+	defer infile.Close()
+
+	var in io.Reader = infile
+	if allowObfuscation && m.Obfuscate {
```

Comment:
Why do you check m.Obfuscate here? If this function is called by `reloadFromDisk` and m.Obfuscate isn't true it won't read the obfuscated version in case it fails to read as non-obfuscated one.